M2/Macaulay2/packages/MultiprojectiveVarieties.m2

```diff
@@ -65,12 +65,14 @@ projectiveVariety Ideal := o -> I -> (
     if flatten m != gens R then error "the given grading on the polynomial ring is not allowed: the degree of each variable must be a standard basis vector of ZZ^r in the commonly used order";
     if not isHomogeneous I then error "expected a (multi)-homogeneous ideal";
     J := I;
-    if o.Saturate 
-    then for x in m do J = saturate(J,ideal x,MinimalGenerators=>o.MinimalGenerators)
-    else if o.MinimalGenerators then J = trim J;
+    if o.Saturate then (
+        if not((J#cache)#?"isMultisaturated" and (J#cache)#"isMultisaturated") then (
+            for x in m do J = saturate(J,ideal x,MinimalGenerators=>o.MinimalGenerators); 
```

Comment:
I have some suggestions to make this cleaner and faster, if you're interested. In particular, `saturate(Ideal, List)` is faster than this and is already cached, so you might not need to do caching on top of it.

Reply:
Also, try using `cacheValue` instead for `projectiveVariety` instead, it'll clean up the code significantly.

Reply:
> I have some suggestions to make this cleaner and faster, if you're interested.

Thanks! Suggestions are always appreciated.

Reply:
> Also, try using `cacheValue` instead for `projectiveVariety` instead, it'll clean up the code significantly.

In future codes I will use the cache and cacheValue much more, but here I made much cache by hand.

Reply:
> In particular, `saturate(Ideal, List)` is faster than this and is already cached, so you might not need to do caching on top of it.

Looking at the code for `saturate(Ideal, List)` it seems to me to be an equivalent way, but this would shorten my code.